### ISSUE TITLE
[UWP] Shaders slang dont works in 1.19

### ISSUE THREAD

**Issue by ErivandoBR** [libretro/RetroArch#16634]
I can't activate shaders slang in Retroarch 1.19
In previous versions 1.17 and 1.18, slang works.

In 1.19, I can only enable shaders if I change the driver to GL. With this driver, I can activate glsl shaders.

- RetroArch UWP 1.19 / Mesa And Angle
- Xbox Series S dev mode

**Comment by hizzlekizzle** [comment]
Can you post a log of it failing to load a slang shader?

**Comment by ErivandoBR** [comment]
> Can you post a log of it failing to load a slang shader?

I tried to generate, but the log file is always blank.

I followed these steps:

-----------
(Step 1)
Inside RetroArch, go to Settings > 
Verbosity - ON
Registration Levels - 0
Record to file - ON
Timestamp - ON
(Step 2)
Go back to the Main Menu in RetroArch and go into Configuration File, save the current config and quit RetroArch.
(Step 3)
Go back into RetroArch and re-create the steps you took to create the error. Once you have done that, Quit RetroArch again, from inside RetroArch.
-----------

After that the log file is generated, but it is always blank.

**Comment by eMSX** [comment]
Same issue here after the update from 1.17.
Error message failed to load shader on the regular d11/d12 driver with every preset..

Xbox series S. Alsof setting it to 4k still resolves in a black screen after loading a core.

**Comment by willekaufhold** [comment]
Same issue here. Slang shaders were working and then just stopped. Now only glsl shaders work. Running on iOS.

**Comment by Dark-Cod3r** [comment]
Yes. facing same issue. and logs also not working. 
Hope someone fix this issue. 
https://www.reddit.com/r/RetroArch/comments/1dikh3l/applying_preset_shader_always_failed_on_xbox_any/

**Comment by joepogo** [comment]
Is there any update on this? I'm holding off on updating until it's fixed. 

Not sure if it's a Xbox or retroarch issue. Can anyone shed some light on that?

**Comment by hizzlekizzle** [comment]
Which build are you guys using? From retroarch.com or from xbox dev store?

**Comment by stholt5233** [comment]
I used previous github build. Still haven't heard anything from anyone about a possible fix.  Luckily the update to 1.19 is nothing major.

**Comment by joepogo** [comment]
@hizzlekizzle I tried both versions as a test and both are broken shader wise. 

I doubt it would be anything shader wise but possibly something more that changed on the retroarch video side code-wise? Since glsl works. 

Hopefully they'll fix it soon. Shaders are a huge part of the experience for a lot of us. 

Hopefully they'll sort the shader issue when using flycast as well.

**Comment by joepogo** [comment]
Just reporting iupdated to the latest version and shaders are still broken. 

I'm currently looking back through commits prior to 1.19.  Does anyone know if there were any major changes to the video code for retroarch before then?

It's obviously not a single shader here or there, it's all of them. I'm wondering if this issue is reproducible through dx11 or 12 on PC, though I think it'd probably would have been reported by now.

**Comment by hizzlekizzle** [comment]
Correct, it's not reproducible on Windows. I don't know of any major changes, no. The lack of logging makes it basically impossible to debug, as well :/

**Comment by EGCdigital** [comment]
Right Here:
https://buildbot.libretro.com/stable/1.17.0/windows-msvc2019-uwp/x64/

**Comment by joepogo** [comment]
Is it possible that the implementation of mesa over angle broke something?

I don't know how that's possible though because the initial test builds of mesa on Xbox retroarch had working slang shaders.

I have a build with mesa working just fine and no slang shaders being broke.

**Comment by hizzlekizzle** [comment]
It's not that no one cares, we just don't have any means to do anything about it. Most of the people who could have moved on to other things, it seems, and we're basically left holding the bag.

We don't even know anything about the Mesa integration.

**Comment by Cracklitrox** [comment]
I see that there is still no solution, it happens to me with the versions given by Git as well as the official RetroArch page, is there any old version where it still works?

**Comment by LibretroAdmin** [comment]
Get a developer in contact with us that can finally untangle this mess.

Either that or we revert everything to do with the Mesa PR since it seems that is the culprit behind all these issues, right?

**Comment by joepogo** [comment]
> Get a developer in contact with us that can finally untangle this mess.
> 
> Either that or we revert everything to do with the Mesa PR since it seems that is the culprit behind all these issues, right?

What's so strange is that we have mesa beta builds on Xbox where the slang shaders *DO  work. 

Something got borked after that with implementation most likely. Hopefully a fix can be made to allow for mesa instead of angle on Xbox retroarch with slang shaders.

**Comment by hizzlekizzle** [comment]
yeah, I'm not totally sure it's a mesa problem, but we'll never know unless someone can build and bisect.

**Comment by worleydl** [comment]
Enabling `HAVE_HLSL` in the preproc options fixed shaders for me.  Here's some experimental vulkan stuff too: https://github.com/worleydl/RetroArch/tree/uwp-updates

**Comment by joepogo** [comment]
Hi @worleydl, 

Thanks so much for fixing this issue! So going forward slang shaders are back to being fixed on mesa and angle builds on Xbox series consoles, correct?

I just had a couple of questions,I noticed on your GitHub repro you reference vulkan on some of your commits, are you looking at attempting to getting vulkan working on Xbox series consoles or am I reading it wrong? :p

One other thing, There's been a long standing issue trying to get threading to work on retroarch mame on series consoles, it was causing crashing before loading on discrete sound core and other games. 

A fix was made to deactivate the threading because of this, but a good number of chd games and other higher end mame games are now not full speed because the threading really helps. 

Is it possible to fix that, it's discussed here with links and a possible fix but it's way over my pay grade. It would be awesome to get threading working and those games back to full speed. Is it possible to get you to take a look at it?

See here:

https://github.com/libretro/mame/issues/326

Regardless, thank you for all you have done. It's difficult to find devs that are willing to fix Xbox related issues. But the good news is that that list is getting shorter!

**Comment by worleydl** [comment]
> So going forward slang shaders are back to being fixed on mesa and angle builds on Xbox series consoles, correct?

Yes

> are you looking at attempting to getting vulkan working on Xbox series consoles

It's "working" via vulkan_dzn which is a limited wrapper over D3D12.  I did experiment with it on ra but I have no plans to push as it currently serves no purpose but to add confusion.  Stability fixes for D3D12 will be coming soon.  dzn may have other applications for basic vulkan renderers, a lot of low level compute is not available on xbox, so it's neat but not very capable.

> Is it possible to get you to take a look at it?

No, that doesn't seem like something I have any ideas for at the moment.

**Comment by joepogo** [comment]
Thank you nonetheless @worleydl for fixing this!

I'll keep an eye out for the next retroarch mess build with fixed shaders. 

Regarding the mesa builds, are all the kinks ironed out or is it still a beta?

I think the build I have is currently the first mesa build the discord had for download. 

Regardless, thank you for your time and answering my questions. 👍😊

**Comment by stholt5233** [comment]
Seems like some work is getting done...but I just checked stable and nightlies on the website...nothing there, 404 Not Found...Does anyone have a link to the UWP release for Xbox series X?

**Comment by eMSX** [comment]
I haven't yet found time to test it but it should be available on the follow-up page,:

https://github.com/EmulationCollective/RetroArch/releases/tag/latest

**Comment by joepogo** [comment]
> I haven't yet found time to test it but it should be available on the follow-up page,:
> 
> https://github.com/EmulationCollective/RetroArch/releases/tag/latest

Is there anywhere else to find this updated mesa build? That link is broken. :/

**Comment by worleydl** [comment]
@DevSergioSiller @joepogo I don't think an official UWP build has been pushed up yet but the EmulationCollective latest does have the patch.  I think they restructured how builds go out so you just have to find the last rolling release instead of the "latest" tag.

I also believe this issue to be resolved now, the previous "ghosts" mentioned were due to stale build configs hanging around.

**Comment by hizzlekizzle** [comment]
Here's a more general link https://github.com/XboxEmulationHub/RetroArch/releases

**Comment by joepogo** [comment]
Thank you @hizzlekizzle and @worleydl! 

I would close this as completed now @ErivandoBR . 

Can confirm everything looks and works great on this build. 👍

**Comment by ErivandoBR** [comment]
@hizzlekizzle @worleydl @joepogo
I tested the version from the link, **slang shaders are working** correctly!
Thank you very much for your work and attention!

**Comment by Cereal-Killa** [comment]
> Here's a more general link https://github.com/XboxEmulationHub/RetroArch/releases

Question, these builds seem to crash on my series S whereas version 18 worked fine, any tips?

- Uninstalled and reinstalled a few times.
- Removed all USB storage devices connected.
- Rebooted XBOX.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/ab905871-8eca-4391-8ff1-e51bf43f29b4" />

**Comment by Cereal-Killa** [comment]
The most recent build that worked for me was 05-26-2025, all newer ones input is frozen after launching:

[Xbox Builds [05-26-2025]](https://github.com/XboxEmulationHub/RetroArch/releases/tag/05-26-2025)

**Comment by Cereal-Killa** [comment]
Yesterday's build fixed it, no more freezing, tks!

https://github.com/XboxEmulationHub/RetroArch/releases/tag/10-14-2025
